vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -158,6 +203,68 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         return dotnetPath;
     }
 
+    private async checkForPartialInstalls(version : string, uninstallLocalRuntime : boolean, uninstallLocalSDK : boolean)
+    {
+        const installingVersions = this.context.extensionState.get<string[]>(this.installingVersionsKey, []);
+        const partialInstall = installingVersions.indexOf(version) >= 0;
+        if (partialInstall)
+        {
+            // Partial install, we never updated our extension to no longer be 'installing'.
+            this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(version));
+
+            // Uninstall everything so we can re-install. For global installs, let the installer handle it.
+            if(uninstallLocalRuntime)
+            {
+                await this.uninstallAll();
+            }
+            else if(uninstallLocalSDK)
+            {
+                await this.uninstallRuntime(version);
+            }
+        }
+    }
+
+    private async acquireGlobalCore(globalInstallerResolver : GlobalInstallerResolver): Promise<string>
+    {
+        const installingVersion = await globalInstallerResolver.getFullVersion();
+        this.checkForPartialInstalls(installingVersion, false, false);
+
+        const installer : IGlobalInstaller = os.platform() === 'linux' ?
+            new LinuxGlobalInstaller(this.context, installingVersion) :
+            new WinMacGlobalInstaller(this.context, installingVersion, await globalInstallerResolver.getInstallerUrl());
+
+        // Indicate that we're beginning to do the install.
+        await this.addVersionToExtensionState(this.installingVersionsKey, installingVersion);
+        this.context.eventStream.post(new DotnetAcquisitionStarted(installingVersion));
+
+        // See if we should return a fake path instead of running the install
+        if(process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH && process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH === 'true')
+        {
+            return 'fake-sdk';
+        }
+
+        const installerResult = await installer.installSDK();
+
+        if(installerResult !== '0')
```

Comment:
nit: feels weird to compare against a string which is really just an int

Reply:
I agree with you. It was initially designed so the exit code could contain strings, but it doesn't look like I use that anywhere. There are definitely a few places where it could be added though to give more verbose error messages. It is inefficient and inelegant, however, I think I will keep it this way for now.